### ISSUE TITLE
VerticalSplitScale.py broken under Python3.5 / matplotlib v2.0.0

### ISSUE THREAD

**Issue by jkrasting** [NOAA-GFDL/MOM6-examples#122]
Calls to axis.set_yscale() are now failing for me. See example in Line 83 of [meridional_overturning.py](https://github.com/NOAA-GFDL/MOM6-examples/blob/4407ab49c95ea6160f0205bf027a29a77528f4b1/tools/analysis/meridional_overturning.py#L83)

e.g.:  
```python
plt.gca().set_yscale('splitscale',zval=[0.,-1000.,-6500.])
```

The calls are failing when I try to save the image to an [io.BytesIO]( https://docs.python.org/3/library/io.html#io.BytesIO) buffer.  I get the following error:

>ValueError: posx and posy should be finite values 
>     args = ('posx and posy should be finite values',) 
>      with_traceback = <built-in method with_traceback of ValueError object>

Oddly enough, similar calls contained within [m6plot.py](https://github.com/NOAA-GFDL/MOM6-examples/blob/4407ab49c95ea6160f0205bf027a29a77528f4b1/tools/analysis/m6plot.py) are not subject to this problem.

I don't see an option for 'splitscale' in the [documentation](http://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.set_yscale.html#matplotlib.axes.Axes.set_yscale) for set_yscale.  I don't see any obvious difference between the calls.

In a possibly related note, the y-axis is flipped for the [zonal_T_bias_WOA05.py](https://github.com/NOAA-GFDL/MOM6-examples/blob/4407ab49c95ea6160f0205bf027a29a77528f4b1/tools/analysis/zonal_T_bias_WOA05.py) script:

![download](https://cloud.githubusercontent.com/assets/6594675/24706436/a43aee66-19dd-11e7-80bc-5895bf5d4803.png)

@adcroft - any ideas?

**Comment by jkrasting** [comment]
@adcroft helped trace this back to the [VerticalSplitScale.py](https://github.com/NOAA-GFDL/MOM6-examples/blob/dev/master/tools/analysis/VerticalSplitScale.py) routine itself in tools/analysis.  Running the file by itself produces the same error.  We still need to figure out why this scale is no longer compatible with the environment.

`python VerticalSplitScale.py`

>lib/python3.5/site-packages/matplotlib/transforms.py:2632: RuntimeWarning: divide by zero encountered in double_scalars
>  y_scale = 1.0 / inh
> ValueError: posx and posy should be finite values

**Comment by jkrasting** [comment]
Issue is independent of backend.  Same error occurs with 'TkAgg', 'WebAgg', and 'MacOSX'.

**Comment by jkrasting** [comment]
One issue appears to be the treatment of vmin and vmax.  In the matplotlib [example to create a custom scale](https://github.com/matplotlib/matplotlib/blob/6030485b99a445bbcda71e1919d2e88e7794d04a/examples/api/plot_custom_scale_example.py#L106), the order of the calls to min and max are reversed in line 106.  Indeed, changing these calls in VerticalSplitScale.py solves this issue and breaks it for prior versions.

I fixed this on the branch `user/jpk/vertical_scale_fix` on my fork of MOM6-examples.  I added a test for the matplotlib version.  This fix is merged into the open pull request #123 for the other analysis script updates.

Here is a [link to the commit](https://github.com/jkrasting/MOM6-examples/commit/c7877daf1ded6ff29bfe4a3fc5b884a30914c3db) on my fork.

Here are two figures using the updated script.  One is from matplotlib v.1.3.3 (same as GFDL PAN) and the other is from v.2.0.0.   It's interesting that the vertical scale for v.1.3.3 bleeds below the x-axis.  This was always the case.

![1_3_1](https://cloud.githubusercontent.com/assets/6594675/24799619/33a66976-1b6a-11e7-8097-90a502338d51.png)

![2_0_0](https://cloud.githubusercontent.com/assets/6594675/24799623/382e45cc-1b6a-11e7-8938-f0619f1343fe.png)
